src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -203,15 +209,54 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
         }
     }
 
+    /**
+     * Returns a since date that is one month before {@code cliUntilDate} (if present) or one month before report
+     *  generation date otherwise.
+     */
+    private static Date getSinceDate(Optional<Date> cliUntilDate) {
+        Calendar cal = Calendar.getInstance();
+        cliUntilDate.ifPresent(cal::setTime);
+        cal.set(Calendar.HOUR_OF_DAY, 0);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        cal.add(Calendar.MONTH, -1);
+        return cal.getTime();
+    }
+
+    /**
+     * Returns date of report generation with time set to 23:59:59.
+     */
+    private static Date getUntilDate() {
```

Comment:
Same for this. 
Rename it to better explain what it does.
Helps with code readability.